src/pages/en/core-concepts/astro-components.md

```diff
@@ -143,6 +143,23 @@ const name = "Astro";
 <MyComponent templateLiteralNameAttribute={`MyNameIs${name}`} />
 ```
 
+:::caution
+HTML element attributes will be converted to strings, so functions and objects don't work predictably when passed as HTML attributes.
+Notably, you can't pass an event handler from an Astro component to an HTML element:
+
+```astro
+---
+function handleClick () {
+    console.log("button clicked!");
+}
+// This doesn't work!
+---
+<button onClick={handleClick}>Nothing will happen when you click me!</button>
```

Comment:
Do you think it isn't necessary to illustrate this gotcha? I felt that it might not be clear enough without an example. I like the idea of showing an alternative, and maybe that will reduce the chance of people trying the first code example if we keep it:


> Notably, you can't assign an event handler to an HTML element in an Astro component:

```astro
---
function handleClick () {
    console.log("button clicked!");
}
// This doesn't work!
---
<button onClick={handleClick}>Nothing will happen when you click me!</button>
```

Instead, use a client-side script to add the event handler:

```astro
<button id="button">Click Me</div>
<script>
function handleClick () {
  console.log("button clicked!");
}
document.getElementById("button").addEventListener("click", handleClick);
</script>
```

Let me know what you think.